src/asami/query.cljc

```diff
@@ -638,11 +638,33 @@
       ;; calculate the aggregates from the final results and project
       (aggregate-over find inner-results))))
 
+(s/defn rewrite-wildcards
+  "In the :where sequence of query replace all occurrences of _ with
+  unique variables."
+  [query]
```

Comment:
When I pull out values from a map, I like to use destructuring. There are a couple of reasons:
- It documents that the argument is a map, and that it has a `:where` key.
- It avoids an explicit `let`/`get` operation. That feels verbose to me.